File: backend/internal/models/social_account.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type SocialAccount struct {
	ID               uuid.UUID  `json:"id"`
	UserID           uuid.UUID  `json:"user_id"`
	Platform         string     `json:"platform"`
	PlatformUserID   *string    `json:"platform_user_id"`
	PlatformUsername *string    `json:"platform_username"`
	AccessToken      string     `json:"-"`
	RefreshToken     *string    `json:"-"`
	TokenExpiresAt   *time.Time `json:"token_expires_at"`
	IsActive         bool       `json:"is_active"`
	CreatedAt        time.Time  `json:"created_at"`
	UpdatedAt        time.Time  `json:"updated_at"`
}

type ConnectAccountRequest struct {
	Platform string `json:"platform" binding:"required"`
	Code     string `json:"code" binding:"required"`
	CodeVerifier string `json:"code_verifier"`
}

File: frontend/lib/models/social_account.dart
class SocialAccount {
  final String id;
  final String platform;
  final String? platformUsername;
  final bool isActive;
  final DateTime createdAt;

  SocialAccount({
    required this.id,
    required this.platform,
    this.platformUsername,
    required this.isActive,
    required this.createdAt,
  });

  factory SocialAccount.fromJson(Map<String, dynamic> json) {
    return SocialAccount(
      id: json['id'] ?? '',
      platform: json['platform'] ?? '',
      platformUsername: json['platform_username'],
      isActive: json['is_active'] ?? false,
      createdAt: DateTime.parse(json['created_at'] ?? DateTime.now().toIso8601String()),
    );
  }
}
